src/OSPSuite.Core/Domain/Services/ParameterValuesCreator.cs

```diff
@@ -96,24 +96,37 @@ public IReadOnlyList<ParameterValue> CreateExpressionFrom(IContainer physicalCon
 
       private IEnumerable<ParameterValue> createExpressionFrom(IContainer container, IReadOnlyList<MoleculeBuilder> molecules) => molecules.SelectMany(x => createExpressionFrom(container, x));
 
-      private void initializeFormulasFor(IReadOnlyList<ParameterValue> parameterValues, IReadOnlyList<ExpressionParameter> expressionParameters)
+      private void updateFromExpression(IReadOnlyList<ParameterValue> parameterValues, IReadOnlyList<ExpressionParameter> expressionParameters)
       {
          var formulaCache = new FormulaCache();
-         parameterValues.Each(formulaTarget =>
+         parameterValues.Each(parameterValue =>
          {
-            var formulaSource = formulaSourceFor(expressionParameters.AllByName(formulaTarget.Name).ToList(), formulaTarget);
+            var expressionParameter = expressionSourceFor(expressionParameters.AllByName(parameterValue.Name).ToList(), parameterValue);
 
-            if (formulaSource == null) 
-               return;
-
-            formulaTarget.Formula = _cloneManager.Clone(formulaSource.Formula, formulaCache);
-            formulaTarget.Value = null;
+            updateFromExpression(expressionParameter, parameterValue, formulaCache);
          });
       }
 
+      private void updateFromExpression(ExpressionParameter expressionParameter, ParameterValue parameterValue, FormulaCache formulaCache)
+      {
+         if (expressionParameter == null)
```

Comment:
Don't we have that exact same code already somehwere. Is this sthg along the line of updating a ParameterValue from another ParameterValue ?